utils/advanced_stats_lib.py

```
@@ -0,0 +1,129 @@
+        
+        return pd.Series(result, index=p_values.index if isinstance(p_values, pd.Series) else None)
+        
+    except Exception as e:
+        logger.error(f"Error applying MCC method '{method}': {e}")
+        # Fallback: return original p-values if correction fails widely
+        return pd.Series(p_vals_arr, index=p_values.index if isinstance(p_values, pd.Series) else None)
+
+
+# --- Collinearity Diagnostics (VIF) ---
+
+def calculate_vif(df: pd.DataFrame, intercept: bool = True) -> pd.DataFrame:
```

Ruff (0.14.10)

Boolean-typed positional argument in function definition

(FBT001)